### PR TITLE
Allow case-insensitive login

### LINKED ISSUE #554: Make usernames case-insensitive
**TL;DR**: Usernames on Lookit should be case-insensitive.

**Narrative**
"As a Participant, I get confused if I registered under e.g. First_last@gmail.com, but then can't log in using my email address first_last@gmail.com. If I request a password reset, I receive it successfully and can reset the password - but then can't actually log in because I've actually reset the password for First_last@gmail.com."

"As a Participant, if I can't log into my account I may figure I didn't really have one and try creating a new account, which may by chance have different capitalization. This can cause errors and it's also confusing to have multiple accounts unintentionally."

We had about 20 duplicate-up-to-case usernames as of several months ago, and I manually cleaned them up on prod (merging/deleting accounts as appropriate). I've handled tech support for 3 participants over 4-5 days (out of ~300) who were having trouble logging in after resetting their password which turned out to be due to the capitalization of their username.

(Note: this affects researchers too, there are just fewer of us and we seem more likely to be using password managers and all-lowercase email addresses for registration.)

**Acceptance Criteria**

- [x] Manual cleanup of staging, prod databases to avoid duplicate accounts
- [x] Migration that safely changes email addresses for duplicate accounts (e.g. adding +<UNIQUESTRING>) so we can fix any that are added right before 
- [x] Add tests for account creation and username changes(don't allow duplicates), for what happens if duplicate account does exist when logging in, and for logging in as user@company.com and UsEr@ComPany.COM 
- [x] QA on new feature: double check we can log in to previously-existing account originally under all-lowercase; can log in to previously-existing account with capital letters; can not change username to an existing address with capitalization changes; can not register new account using existing address with capitalization changes; can send email to participant; can trigger password reset email, reset password, and log in successfully

**Implementation Notes**
Could have migration actually try to handle merging accounts (e.g., move all children to acct 1, all messages/logs/etc. reference acct 1) and then match children manually afterwards, but I think that'll be substantially more trouble than it's worth to test carefully enough, versus just handling the relatively few cases manually.

### PR DESCRIPTION
**Pull request by Datamance**
Fixes #554

### CODE REVIEW

File: accounts/tests.py [hunk 1]
```diff
@@ -250,6 +266,25 @@ def test_lab_researcher_can_create_study(self):
     def test_unaffiliated_researcher_cannot_create_study(self):
         self.assertFalse(self.unaffiliated_researcher.can_create_study())
 
+    def test_create_user_lowercases_username(self):
+        # TODO: Do we actually use `create_user` anywhere?
```
Comment (kimberscott) [inline accounts/tests.py:270]: only in create_superuser - otherwise users are created via the ParticipantSignupForm & ResearcherRegistrationForm. Ok for me to push addition of clean_username methods to those on this branch?

File: accounts/models.py [hunk 2]
```diff
@@ -42,11 +42,16 @@
...
+        return self.get(**{self.model.USERNAME_FIELD + "__iexact": username})
+
     def create_user(self, username, password=None):
         if not username:
             raise ValueError("Users must have a username")
 
-        user = self.model(username=self.normalize_email(username), is_active=True)
+        user = self.model(
+            username=self.normalize_email(username).lower(), is_active=True
+        )
```
Comment (kimberscott) [inline accounts/models.py:54]: Like you noted below we don't usually use this - to do the equivalent when people create/update accounts, we'd want to handle that in the ParticipantSignupForm, ResearcherRegistrationForm, and AccountUpdateForm

File: accounts/tests.py (hunk 1 above)
Comment (Datamance) [reply accounts/tests.py:270]: Which branch are those currently on?

File: accounts/tests.py [hunk 3]
```diff
@@ -250,6 +266,25 @@ def test_lab_researcher_can_create_study(self):
...
+        # Test that login is case-insensitive
+        response = self.client.post(
+            reverse("login"),
+            {
+                "username": "mixedcase@gmail.com",
+                "password": self.test_password,
+                "next": "/",
+            },
+            follow=True,
+        )
```
Comment (kimberscott) [inline accounts/tests.py:284]: should also check we can log in by posting a not-all-lowercase email

File: accounts/forms.py [hunk 4]
```diff
@@ -179,6 +207,7 @@ def __init__(self, *args, **kwargs):
 
     class Meta:
         model = User
+        field_classes = {"username": ForceLowercaseEmailField}
```
Comment (kimberscott) [inline accounts/forms.py:210]: how come we need this here but not in ResearcherRegistrationForm?

Comment (Datamance) [reply accounts/forms.py:210]: Meta is inheriting from `LowercaseUsernameUserCreationForm.Meta` in the other cases

**Review by kimberscott — APPROVED** [review]
LGTM! Thanks for handling this so fast and for adding the tests. Fingers crossed we avoid a bunch of confusion for participants here!
